bug: hospital-discharge-management fails to compile — initialize() calls undefined save_hospital_registry instead of set_hospital_registry

## Summary

`contracts/hospital-discharge-management/src/lib.rs:48` calls `save_hospital_registry(&env, &hospital_registry)`, but no function with that name exists in `storage.rs` — the actual setter is `set_hospital_registry` (`storage.rs:108-110`). Confirmed with `cargo check -p hospital-discharge-management`: `error[E0425]: cannot find function 'save_hospital_registry' in this scope`.

## Risk

The crate does not compile at all, so `initialize` — the entry point every other function in the contract depends on via `get_hospital_registry` — has never actually run.

## Acceptance Criteria

- [ ] Change the call in `initialize` to `set_hospital_registry(&env, &hospital_registry)`
- [ ] `cargo check -p hospital-discharge-management` and `cargo test -p hospital-discharge-management` pass
- [ ] Add a regression test asserting `initialize` followed by any state-mutating call succeeds

--- a/contracts/hospital-discharge-management/src/lib.rs
+++ b/contracts/hospital-discharge-management/src/lib.rs
@@ -45,7 +45,7 @@ impl HospitalDischargeContract {
     /// Initialize the contract with a hospital registry address
     pub fn initialize(env: Env, hospital_registry: Address) -> Result<(), Error> {
         hospital_registry.require_auth();
-        save_hospital_registry(&env, &hospital_registry);
+        set_hospital_registry(&env, &hospital_registry);
         Ok(())
     }
 

--- a/contracts/hospital-discharge-management/src/test.rs
+++ b/contracts/hospital-discharge-management/src/test.rs
@@ -445,6 +445,29 @@ fn test_full_discharge_workflow() {
 }
 
 #[test]
+fn test_initialize_then_discharge_planning() {
+    let (env, admin, _patient, patient_id, hospital_id) = create_test_env();
+    let contract_id = env.register(HospitalDischargeContract, ());
+    let client = HospitalDischargeContractClient::new(&env, &contract_id);
+
+    // Initialize the contract with the hospital registry
+    client.initialize(&admin);
+
+    // Verify initialize succeeded by running a state-mutating call
+    let plan_id = client.initiate_discharge_planning(
+        &admin,
+        &patient_id,
+        &hospital_id,
+        &1000u64,
+        &2000u64,
+    );
+
+    assert_eq!(plan_id, 1);
+    let plan = client.get_discharge_plan(&plan_id);
+    assert_eq!(plan.patient_id, patient_id);
+}
+
+#[test]
 fn test_complete_discharge_unauthorized() {
     let (env, admin, _patient, patient_id, hospital_id) = create_test_env();
     let contract_id = env.register(HospitalDischargeContract, ());

